The previous test failure was not related `FAILED tests/components/wemo/test_binary_sensor.py::TestInsight::test_async_update_locked_multiple_callbacks[pyloop]`

--- a/homeassistant/helpers/integration_platform.py
+++ b/homeassistant/helpers/integration_platform.py
@@ -29,12 +29,12 @@ async def _async_process_single_integration_platform(
     hass: HomeAssistant, component_name: str, integration_platform: IntegrationPlatform
 ) -> None:
     """Process a single integration platform."""
-    platform_name = integration_platform.platform_name
     if component_name in integration_platform.seen_components:
         return
     integration_platform.seen_components.add(component_name)
 
     integration = await async_get_integration(hass, component_name)
+    platform_name = integration_platform.platform_name
 
     try:
         platform = integration.get_platform(platform_name)

--- a/tests/components/intent/test_init.py
+++ b/tests/components/intent/test_init.py
@@ -60,6 +60,7 @@ async def test_cover_intents_loading(hass):
         )
 
     assert await async_setup_component(hass, "cover", {})
+    await hass.async_block_till_done()
 
     hass.states.async_set("cover.garage_door", "closed")
     calls = async_mock_service(hass, "cover", SERVICE_OPEN_COVER)
@@ -81,6 +82,7 @@ async def test_turn_on_intent(hass):
     """Test HassTurnOn intent."""
     result = await async_setup_component(hass, "homeassistant", {})
     result = await async_setup_component(hass, "intent", {})
+    await hass.async_block_till_done()
     assert result
 
     hass.states.async_set("light.test_light", "off")

--- a/tests/components/lovelace/test_system_health.py
+++ b/tests/components/lovelace/test_system_health.py
@@ -24,6 +24,7 @@ async def test_system_health_info_storage(hass, hass_storage):
         "data": {"config": {"resources": [], "views": []}},
     }
     assert await async_setup_component(hass, "lovelace", {})
+    await hass.async_block_till_done()
     info = await get_system_health_info(hass, "lovelace")
     assert info == {"dashboards": 1, "mode": "storage", "resources": 0, "views": 0}
 
@@ -32,6 +33,7 @@ async def test_system_health_info_yaml(hass):
     """Test system health info endpoint."""
     assert await async_setup_component(hass, "system_health", {})
     assert await async_setup_component(hass, "lovelace", {"lovelace": {"mode": "YAML"}})
+    await hass.async_block_till_done()
     with patch(
         "homeassistant.components.lovelace.dashboard.load_yaml",
         return_value={"views": [{"cards": []}]},
@@ -44,6 +46,7 @@ async def test_system_health_info_yaml_not_found(hass):
     """Test system health info endpoint."""
     assert await async_setup_component(hass, "system_health", {})
     assert await async_setup_component(hass, "lovelace", {"lovelace": {"mode": "YAML"}})
+    await hass.async_block_till_done()
     info = await get_system_health_info(hass, "lovelace")
     assert info == {
         "dashboards": 1,

